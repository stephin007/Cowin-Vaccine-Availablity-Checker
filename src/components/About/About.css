.about-container {
    max-width: 1100px;
    margin: auto;
    overflow: hidden;
    padding: 0 2rem;
    font-family: "Fira Code Medium", "monospace";
}

.about-head{
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 10px;
}

.about-head p{
   font-weight: bold;
    font-size: large;
}

h4{
    color: #333333;
    text-transform: capitalize;
}

@media screen and (max-width: 500px) {
    .about-container {
        width: 350px;
    }

    .about-head p{
        font-weight: bold;
        font-size: 13px;
    }

    .about-head h1{
        font-weight: bold;
        font-size: 20px;
    }

    h4{
        font-size: 10px;
    }
}
