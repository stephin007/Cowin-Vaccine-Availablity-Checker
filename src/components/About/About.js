import { useState, useEffect } from "react";
import { CircularProgress } from "@material-ui/core";

import "./About.css";

const About = () => {
  const [contributors, setContributors] = useState([]);
  const [contributorCount, setContributorCount] = useState(0);
  const [loading, setLoading] = useState(true);

  const getContributors = async () => {
    fetch(
      "https://api.github.com/repos/stephin007/Cowin-Vaccine-Availablity-Checker/contributors"
    )
      .then((response) => response.json())
      .then((data) => {
        setContributors(data);
        console.log(data);
        setContributorCount(data.length);
        setLoading(false);
      });
  };

  useEffect(() => {
    getContributors();
  }, []);

  return (
    <>
      <div className="about-container">
        <div className="about-head">
          <h1>About the App</h1>
        </div>
        <hr />
        <br />
        <div className="about_section">
          <div className="about_section_left">
            <div className="about_section_main_heading">
              <h4>
                App to checkout the latest COVID19 Vaccination Slots Across
                IndiaðŸŽ¨.
              </h4>
            </div>
            <br />
            <div className="about_section_subheadings">
              <h3>Details about the vaccine that you will find here</h3>
              <ul>
                <li>Name of the Vaccine</li>
                <li>Whether paid or free</li>
              </ul>
              <h3>Highlight features of the app</h3>
              <ul>
                <li> No more OTP!</li>
                <li>
                  You can check for slots before hand and book your slot on the
                  Aarogya Setu App accordingly
                </li>
                <li>
                  You can check for slots across all districts of India, even
                  using postal codes
                </li>
                <li> Details about free and paid vaccines is also available</li>
                <li>
                  Fully open sourced information made on the latest technology
                  stack
                </li>
                <li> We also have a feature of Map*</li>
              </ul>
              <p>
                *This feature is in Beta Version, proper coordinates may not be
                available. Also this feature is only available in the desktop
                version
              </p>
              <h3>Whatâ€™s Next?</h3>
              <ul>
                <li>
                  Notification feature to give you real time alerts when slots
                  are available
                </li>
              </ul>
            </div>
          </div>
          <div className="about_section_right">
            <div className="contributor_text">
              <h3>
                CONTRIBUTORS <span>{contributorCount}</span>
              </h3>
            </div>
            {!loading ? (
              contributors.map((contributor) => {
                const { contributions, avatar_url, login, type, html_url } =
                  contributor;
                return (
                  <div className="contributors_block">
                    <div className="contributor_individual">
                      <div className="contributor_image">
                        <img src={avatar_url} alt="contributor avatar" />
                      </div>
                      <div className="contributor_detail">
                        <a href={html_url}>{login}</a>
                        <div className="contribution_count">
                          <p>No of Contributions : </p>
                          <strong>{contributions}</strong>
                        </div>
                        <div className="contributor_type">
                          {login === "stephin007" ||
                          login === "Justinnn07" ||
                          login === "wise-introvert" ? (
                            <p>Maintainer</p>
                          ) : (
                            <p>{type}</p>
                          )}
                        </div>
                      </div>
                    </div>
                  </div>
                );
              })
            ) : (
              <>
                <div
                  style={{
                    display: "flex",
                    flexDirection: "column",
                    alignItems: "center",
                    justifyContent: "center",
                    textAlign: "center",
                    width: "auto",
                    height: "100vh",
                  }}
                >
                  <CircularProgress />
                </div>
              </>
            )}
          </div>
        </div>
      </div>
    </>
  );
};

export default About;
