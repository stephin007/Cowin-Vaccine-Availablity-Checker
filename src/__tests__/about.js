import { shallow } from 'enzyme';
import React from 'react';
import About from '../components/About/About';

describe('render about component', () => {
	const wrapper = shallow(<About />);

	it('render about properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('render head properly', () => {
		expect(wrapper.find('.about-head').text()).toEqual('About the App');
	});
});
