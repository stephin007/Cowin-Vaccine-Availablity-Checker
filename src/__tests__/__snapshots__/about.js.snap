// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render about component render about properly 1`] = `ShallowWrapper {}`;
